src/Desktop/UIAutomation/CustomObjects/CustomProperty.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+
+namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
+{
+    public class CustomProperty
+    {
+        /// <summary>The RFC4122 globally unique identifier of this property.</summary>
+        [JsonProperty("guid")]
+        #pragma warning disable CA1720 // Identifier contains type name: name from JSON
+        public Guid? Guid { get; set; }
+        #pragma warning restore CA1720 // Identifier contains type name: name from JSON
+        /// <summary>A textual description of this property.</summary>
+        [JsonProperty("programmaticName")]
+        public string ProgrammaticName { get; set; }
+
+        /// <summary>The data type of this property's value as specified by the user, one of string, int, bool, double, point, or element.</summary>
+        // TODO Bill: add enum (with values member)
+        [JsonProperty("uiaType")]
+        public string UserType { get; set; }
+
+        /// <summary>The dynamic ID assigned to this property by the system.</summary>
+        [JsonIgnore]
+        public int DynamicId { get; set; }
+
+        internal void Validate()
+        {
+            if (this.Guid == null) throw new ArgumentException("Missing GUID in custom property definition.");
+            if (this.ProgrammaticName == null) throw new ArgumentException("Missing programmatic name in custom property definition.");
+            if (this.UserType == null) throw new ArgumentException("Missing type in custom property definition.");
+            switch (this.UserType)
```

Comment:
How about breaking this switch out into a separate function that either validates the type or returns the Type (note the capital T in "Type").